### PR TITLE
Chore: Update linting configuration

### PR DESCRIPTION
**Pull request by ModeSevenIndustrialSolutions**
Bump action versions, implement additional checks.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,49 +1,103 @@
...
+
 ci:
   autofix_commit_msg: "Chore: pre-commit autoupdate"
 
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: f71fa2c1f9cf5cb705f73dffe4b21f7c61470ba9 # frozen: v4.4.0
+    rev: cef0300fd0fc4d2a87a85fa2093c6b283ea36f4b # frozen: 5.0.0
     hooks:
-      - id: trailing-whitespace
```
Comment (askb) [inline .pre-commit-config.yaml:9]: This is required.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,49 +1,103 @@
...
+      - id: name-tests-test
+        args: ["--pytest-test-first"]
       - id: no-commit-to-branch
         args:
           - --branch=dev
           - --branch=master
           - --branch=main
           - --branch=rc
           - --branch=production
+      # - id: pretty-format-json
```
Comment (askb) [inline .pre-commit-config.yaml:47]: Please remove commented tests if we dont need them

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,49 +1,103 @@
...
+      - id: check-shebang-scripts-are-executable
+      - id: check-symlinks
+      - id: check-toml
       - id: check-xml
       - id: check-yaml
       - id: debug-statements
+      - id: destroyed-symlinks
+      - id: detect-aws-credentials
+        args: ["--allow-missing-credentials"]
+      # - id: double-quote-string-fixer
```
Comment (askb) [inline .pre-commit-config.yaml:29]: Please remove

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,49 +1,103 @@
...
+      - id: detect-private-key
       - id: end-of-file-fixer
-      - id: requirements-txt-fixer
+      # - id: file-contents-sorter
+      # - id: fix-byte-order-marker
+      # - id: forbid-new-submodules
+      # - id: forbid-submodules
```
Comment (askb) [inline .pre-commit-config.yaml:35]: Please remove lines

File: .pre-commit-config.yaml (hunk 2 above)
Comment (askb) [inline .pre-commit-config.yaml:47]: Please remove

**Review by askb — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ModeSevenIndustrialSolutions) [reply .pre-commit-config.yaml:9]: It's there :-)

**Review by askb — APPROVED** [review]

**Pull request merged**